pex/resolver.py

```diff
@@ -485,7 +486,16 @@ def finalize_install(self, install_requests):
         # pex:   * - paths that do not exist or will be imported via zipimport
         # pex.pex 2.0.2
         #
-        wheel_dir_hash = fingerprint_path(self.install_chroot)
+        cached_fingerprint = None  # type: Optional[str]
+        try:
+            installed_wheel = InstalledWheel.load(self.install_chroot)
+        except LoadError:
+            # We support legacy chroots below by calculating the chroot fingerprint just in time.
+            pass
+        else:
+            cached_fingerprint = installed_wheel.fingerprint
```

Comment:
@zmanji so this represents the avoidance of ~3s to hash PyTorch 1 of 2 times. It only works for new caches though (`rm -rf ~/.pex/installed_wheels`). I wanted to keep the status quo here with the chroot symlinking and just get this easy win for now. I do have some notes drawn up revisiting using a separate chroot hash, so this may all go away later, but with the same net effect of skipping the expensive chroot hashing.